drivers/gles2/rasterizer_storage_gles2.cpp

```diff
@@ -6069,6 +6069,13 @@ void RasterizerStorageGLES2::initialize() {
 	config.rgtc_supported = config.extensions.has("GL_EXT_texture_compression_rgtc") || config.extensions.has("GL_ARB_texture_compression_rgtc") || config.extensions.has("EXT_texture_compression_rgtc");
 	config.bptc_supported = config.extensions.has("GL_ARB_texture_compression_bptc") || config.extensions.has("EXT_texture_compression_bptc");
 
+// Skip depth texture check for webgl 1.0
+#ifdef JAVASCRIPT_ENABLED
+	config.depth_internalformat = GL_DEPTH_COMPONENT16;
```

Comment:
Please look at how these variables are defined earlier in ``initialize()``.

You are overwriting variables that have valid inputs. For example, ``support_depth_texture`` is always set to false, even though it works when the browser has the ``WEBGL_depth_texture`` extension. 

Further ``depth_type`` gets set correctly in the branch that you have compiled out. 

Im guessing that the correct fix will be less intrusive. Have you tried just adding a condition for JAVASCRIPT_ENABLED at line 6109-6114

Reply:
So something like this?

```
#if defined(GLES_OVER_GL) || defined(JAVASCRIPT_ENABLED)
			config.depth_internalformat = GL_DEPTH_COMPONENT16;
#else
			// OES_depth_texture extension only specifies GL_DEPTH_COMPONENT.
			config.depth_internalformat = GL_DEPTH_COMPONENT;
#endif
```

If so, that branch never gets executed on my system. Inserting some debug prints shows that...

```
		if (status != GL_FRAMEBUFFER_COMPLETE) {
```
Evaluates to non-zero/true. (status is 0x8CD5/GL_FRAMEBUFFER_COMPLETE)

![image](https://user-images.githubusercontent.com/71855294/105990271-dcf6ec80-6067-11eb-8c83-794df8e3c64f.png)
![image](https://user-images.githubusercontent.com/71855294/105990584-5098f980-6068-11eb-8055-38cc9dbee691.png)

Reply:
So starting from scratch with this I came with something that might be better...

https://github.com/WeaselShoppe/godot/commit/5ea36e836d0df0dd72b891a6d678e3809a6818ee

The issues that I understand:

-config->depth_internalformat should probably be set to GL_DEPTH_COMPONENT16 at the beginning of RasterizerStorageGLES2::initialize if JAVASCRIPT_ENABLED

-config->depth_internalformat is passed to glTexImage2D in several places in the GLES2 rendering code. If this is equal to GL_DEPTH_COMPONENT16 that will cause errors in WebGL since GL_DEPTH_COMPONENT16 is not a valid parameter to pass.

The branch I pushed just now should solve both problems. no rendering errors and no warnings in console. Seems minimally intrusive too. Not sure if this is right approach though?

![image](https://user-images.githubusercontent.com/71855294/105999313-2567d780-6073-11eb-9882-ade9a0393c3d.png)

Reply:
Why not just replace line 5970 with
```
#ifdef JAVASCRIPT_ENABLED
    config.depth_internalformat = GL_DEPTH_COMPONENT16;
#else
    config.depth_internalformat = GL_DEPTH_COMPONENT;
#endif
```

Reply:
If we do that, it will cause failures elsewhere in the code where config.depth_internalformat is passed to glTexImage2D (GL_DEPTH_COMPONENT16 is not a valid format for this on WebGL1.0)

I have a simple fix here that should be less intrusive. Does this look like it might be more appropriate?
https://github.com/WeaselShoppe/godot/commit/5ea36e836d0df0dd72b891a6d678e3809a6818ee

Reply:
Yes, I looked at your branch when you posted it before. You are replacing the config variable with a global macro in two different files. The problem with a hardcoded approach like this is that it is hard to maintain. When coming up with a solution it helps to write your solution as if the next person who works on this code will be learning it for the first time. In that case it is much more preferable for them to be able to look through ``initialize`` and understand how the config variables are determined. Rather than having the config determined for some platforms in ``initialize`` and in a global macro for others.